phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -1774,11 +1778,20 @@ public QueryLogger createQueryLogger(CompilableStatement stmt, String sql) throw
                 }
             }
         }
-        QueryLogger queryLogger = QueryLogger.getInstance(connection,isSystemTable);
-        QueryLoggerUtil.logInitialDetails(queryLogger, connection.getTenantId(),
+
+        QueryLogger queryLogger = QueryLogger.createInstance(connection,isSystemTable,
+            isCriticalStatement(stmt));
+        QueryLoggerUtil.logInitialDetails(queryLogger, connection.getTenantId(), 
                 connection.getQueryServices(), sql, getParameters());
+        if (isCriticalStatement(stmt)) {
+            queryLogger.sync(null, null);
+        }
         return queryLogger;
     }
+
+    private boolean isCriticalStatement(CompilableStatement stmt) {
```

Comment:
On a second thought, How about logging every DML that is not a "Update statement"? 
It might be an overkill if we have people creating tons of views though, but we could always drop those as well.

Reply:
Also, as discussed offline, we might not want to log `DropTableStatement` which are actually `DROP VIEW` since we can have lot of those.